src/routes/safe/components/Apps/components/AppFrame.tsx

```diff
@@ -52,6 +54,22 @@ const AppFrame = forwardRef<HTMLIFrameElement, AppFrameProps>(function AppFrameC
   const history = useHistory()
   const { consentReceived, onConsentReceipt } = useLegalConsent()
   const redirectToBalance = () => history.push(`${SAFELIST_ADDRESS}/${safeAddress}/balances`)
+  const timer = useRef<number>()
+  const [appTimeout, setAppTimeout] = useState(false)
+
+  useEffect(() => {
+    if (appIsLoading) {
+      timer.current = setTimeout(() => {
+        setAppTimeout(true)
+      }, TIMEOUT)
+    } else {
+      clearTimeout(timer.current)
```

Comment:
does it make sense to bolster with `setAppTimeout(false)`

Reply:
Yes make sense, I did it, thanks!